src/Microsoft.IdentityModel.Clients.ActiveDirectory/Features/CertificateAssertion/ClientAssertionCertificate.cs

```diff
@@ -64,16 +65,33 @@ public ClientAssertionCertificate(string clientId, X509Certificate2 certificate)
 	            if (certificate.PublicKey.Key.KeySize < MinKeySizeInBits)
 #endif	
             {
-                throw new ArgumentOutOfRangeException("certificate",
+                throw new ArgumentOutOfRangeException(nameof(certificate),
                     string.Format(CultureInfo.InvariantCulture, AdalErrorMessage.CertificateKeySizeTooSmallTemplate, MinKeySizeInBits));
             }
 
             this.clientId = clientId;
             this.Certificate = certificate;
         }
 
+		/// <summary>
+		/// Constructor to create credential with client Id and certificate and claimSet.
+		/// </summary>
+		/// <param name="clientId">Identifier of the client requesting the token.</param>
+		/// <param name="certificate">The certificate used as credential.</param>
+		/// <param name="additionalClaims">The additional claims to be added to the payload of the client assertion.</param>
+		public ClientAssertionCertificate(string clientId, X509Certificate2 certificate, Dictionary<string, string> additionalClaims) : this(clientId, certificate)
+	    {
+		    this.AdditionalClaims = additionalClaims ?? throw new ArgumentNullException(nameof(additionalClaims));
+	    }
 
-        /// <summary>
+		// TODO: Consider if we should have a public setter on AdditionalClaims and perhaps remove the additional contructor.
+
+		/// <summary>
+		/// Gets the additional claims to be added to the payload of the ClientAssertion.
+		/// </summary>
+	    public Dictionary<string, string> AdditionalClaims { get; private set; }
```

Comment:
Consider returning IDictionary<string, string> rather than Dictionary<string,string>

Reply:
An advantage with the ctor allows one to have one call.